services/datamanager/builtin/builtin.go

```diff
@@ -539,30 +544,33 @@ func (svc *builtIn) uploadData(cancelCtx context.Context, intervalMins float64)
 			case <-svc.syncTicker.C:
 				svc.lock.Lock()
 				if svc.syncer != nil {
-					svc.sync(cancelCtx)
+					// If selective sync is not enabled (false), we default to not sync until the trigger
```

Comment:
Perhaps just `If selective sync is disabled, sync. If it is enabled, check the condition below.`

Reply:
Updated.